src/Executor/IniSetter.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Runtime\Executor;
+
+use RuntimeException;
+
+class IniSetter implements RuntimeExecutor
+{
+    /**
+     * @var array{}|array{
+     *   string: array{
+     *     value: bool|float|int|string,
+     *     package: string
+     *   }
+     * }
+     */
+    private $alreadySet = [];
+
+    /**
+     * @param RuntimeOptions $options
+     */
+    public function execute(string $projectDir, array $options): void
+    {
+        foreach ($options as $package => $packageOptions) {
+            foreach ($packageOptions['ini']['set'] ?? [] as $key => $value) {
+                if ($value === null) {
+                    continue;
+                }
+                if (is_scalar($value) === false) {
+                    throw new RuntimeException(
+                        "[atoolo.runtime.init.set]: "
+                        . "Value for $key in package $package must be scalar"
+                    );
+                }
+                $this->initSet($package, $key, $value);
+            }
+        }
+    }
```

Comment:
No, there is no hierarchy here.  The aim is to prevent two packages from requiring different values. If this is the case, a RuntimeException is thrown and the "conflict" must be resolved by correcting this in the packages.